hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/event/EventPublisherHolder.java

```diff
@@ -9,63 +9,158 @@
  */
 package org.eclipse.hawkbit.repository.event;
 
-import lombok.AccessLevel;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.hawkbit.repository.event.remote.AbstractRemoteEvent;
+import org.eclipse.hawkbit.repository.event.remote.CancelTargetAssignmentEvent;
+import org.eclipse.hawkbit.repository.event.remote.MultiActionCancelEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingCancelTargetAssignmentEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingMultiActionAssignEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingMultiActionCancelEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingTargetAssignDistributionSetEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingTargetAttributesRequestedEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingTargetCreatedEvent;
+import org.eclipse.hawkbit.repository.event.remote.ProcessingTargetDeletedEvent;
+import org.eclipse.hawkbit.repository.event.remote.MultiActionAssignEvent;
+import org.eclipse.hawkbit.repository.event.remote.TargetAssignDistributionSetEvent;
+import org.eclipse.hawkbit.repository.event.remote.TargetAttributesRequestedEvent;
+import org.eclipse.hawkbit.repository.event.remote.TargetDeletedEvent;
+import org.eclipse.hawkbit.repository.event.remote.entity.TargetCreatedEvent;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.cloud.bus.BusProperties;
-import org.springframework.cloud.bus.ServiceMatcher;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.context.ApplicationEvent;
 import org.springframework.context.ApplicationEventPublisher;
 
-/**
- * A singleton bean which holds the event publisher and service origin id in order to publish remote application events.
- * It can be used in beans not instantiated by spring e.g. JPA entities which cannot be auto-wired.
- */
-@NoArgsConstructor(access = AccessLevel.PRIVATE)
-@SuppressWarnings("java:S6548") // java:S6548 - singleton holder ensures static access to spring resources in some places
+@Slf4j
 public final class EventPublisherHolder {
 
-    private static final EventPublisherHolder SINGLETON = new EventPublisherHolder();
+    @Value("${org.eclipse.hawkbit.events.remote-enabled:false}")
```

Comment:
shouldn't we have remote enabled by default? for all microservices it shall be true?
only disabled in monolith?